cloud/linode/loadbalancers.go

```diff
@@ -246,55 +249,36 @@ func (l *loadbalancers) getNodeBalancerDeviceId(ctx context.Context, firewallID,
 	return 0, false, nil
 }
 
-// updateNodeBalancerFirewall updates the firewall attached to the nodebalancer
-//
-// Firewall is updated in 3 scenario's:
-// - Add new firewall to the nodeBalancer
-// - Update firewall attached to the nodebalancer
-// - Remove firewall attached
-func (l *loadbalancers) updateNodeBalancerFirewall(ctx context.Context, service *v1.Service, nb *linodego.NodeBalancer) error {
+func (l *loadbalancers) updateFWwithID(ctx context.Context, service *v1.Service, nb *linodego.NodeBalancer) error {
 	var newFirewallID int
 	var err error
 
-	// get the new firewall id form the annotaiton (if any).
-	fwid, ok := service.GetAnnotations()[annLinodeCloudFirewallID]
-	if ok {
-		newFirewallID, err = strconv.Atoi(fwid)
-		if err != nil {
-			return err
-		}
+	fwID := service.GetAnnotations()[annLinodeCloudFirewallID]
+	newFirewallID, err = strconv.Atoi(fwID)
+	if err != nil {
+		return err
 	}
 
-	// get the list of attached firewalls to the node-balancer.
+	// See if a firewall is attached to the nodebalancer first.
 	firewalls, err := l.client.ListNodeBalancerFirewalls(ctx, nb.ID, &linodego.ListOptions{})
 	if err != nil {
-		if !ok {
-			if err.Error() != "[404] Not Found" {
-				return err
-			} else {
-				return nil
-			}
-		}
+		return err
 	}
-
-	// if there are no attached firewalls and no annotation added; return.
-	if !ok && len(firewalls) == 0 {
-		return nil
+	if len(firewalls) > 1 {
+		return errTooManyFirewalls
 	}
 
-	// get the ID of the firewall that is already attached to the nodeBalancer.
+	// get the ID of the firewall that is already attached to the nodeBalancer, if we have one.
 	var existingFirewallID int
-	if len(firewalls) != 0 {
+	if len(firewalls) == 1 {
 		existingFirewallID = firewalls[0].ID
 	}
 
-	// if existing firewall and new firewall differs, update accordingly.
+	// if existing firewall and new firewall differs,
```

Comment:
trailing comma... maybe just write out what happens now, otherwise the comment is not super helpful